src/Infrastructure/Search/FilterableTermsUpdater.php

```diff
@@ -403,7 +403,11 @@ private function isHierarchicalProperty(
         MultilingualString|MultilingualLink|AdmissionRequirement|Degree $flatProperty
     ): bool {
 
-        return $flatProperty instanceof Degree || $flatProperty instanceof AdmissionRequirement;
+        return in_array($flatProperty::class, [
+            Degree::class,
+            AdmissionRequirement::class,
+            MultilingualLink::class,
+        ], true);
```

Comment:
It seems only `MultilingualString` is not hierarchical (yet), but a positive check looks positive :)